lti_consumer/lti_consumer.py

```diff
@@ -0,0 +1,839 @@
+        """
+        return unicode(self.course_id)  # pylint: disable=no-member
+
+    @property
+    def role(self):
+        """
+        Get system user role and convert it to LTI role.
+        """
+        return ROLE_MAP.get(self.runtime.get_user_role(), u'Student')
+
+    @property
+    def course(self):
+        """
+        Return course by course id.
+        """
+        return self.runtime.descriptor_runtime.modulestore.get_course(self.course_id)  # pylint: disable=no-member
+
+    @property
+    def lti_provider_key_secret(self):
+        """
+        Obtains client_key and client_secret credentials from current course.
+        """
+        for lti_passport in self.course.lti_passports:
+            try:
+                lti_id, key, secret = [i.strip() for i in lti_passport.split(':')]
+            except ValueError:
+                _ = self.runtime.service(self, "i18n").ugettext
+                msg = _('Could not parse LTI passport: {lti_passport}. Should be "id:key:secret" string.').format(
+                    lti_passport='{0!r}'.format(lti_passport)
+                )
+                raise LtiError(msg)
+
+            if lti_id == self.lti_id.strip():
+                return key, secret
+
+        return '', ''
+
+    @property
+    def user_id(self):
+        """
+        Returns the opaque anonymous_student_id for the current user.
+        """
+        user_id = self.runtime.anonymous_student_id
+        if user_id is None:
+            raise LtiError("Could not get user id for current request")
+        return unicode(urllib.quote(user_id))
+
+    @property
+    def resource_link_id(self):
+        """
+        This is an opaque unique identifier that the LTI Tool Consumer guarantees will be unique
+        within the Tool Consumer for every placement of the link.
+
+        If the tool / activity is placed multiple times in the same context,
+        each of those placements will be distinct.
+
+        This value will also change if the item is exported from one system or
+        context and imported into another system or context.
+
+        resource_link_id is a required LTI launch parameter.
+
+        Example:  u'edx.org-i4x-2-3-lti-31de800015cf4afb973356dbe81496df'
+
+        Hostname, edx.org,
+        makes resource_link_id change on import to another system.
+
+        Last part of location, location.name - 31de800015cf4afb973356dbe81496df,
+        is random hash, updated by course_id,
+        this makes resource_link_id unique inside single course.
+
+        First part of location is tag-org-course-category, i4x-2-3-lti.
+
+        Location.name itself does not change on import to another course,
+        but org and course_id change.
+
+        So together with org and course_id in a form of
+        i4x-2-3-lti-31de800015cf4afb973356dbe81496df this part of resource_link_id:
+        makes resource_link_id to be unique among courses inside same system.
+        """
```

Comment:
The docstring above refers to an old-style, deprecated block locator:
https://github.com/edx/opaque-keys/blob/master/opaque_keys/edx/locator.py#L625
Will the scheme described above be affected by a new-style `block-v1` tag? Will LTI use this new-style tag at all?